kwc-challenge-set-listing.html

```diff
@@ -8,27 +9,80 @@
 -->
 
 <dom-module id="kwc-challenge-set-listing">
-  <template>
-    <style>
-      :host {
-        display: block;
-      }
-    </style>
-    <h2>Hello [[prop1]]!</h2>
-  </template>
-
-  <script>
-    Polymer({
-
-      is: 'kwc-challenge-set-listing',
-
-      properties: {
-        prop1: {
-          type: String,
-          value: 'kwc-challenge-set-listing',
-        },
-      },
-
-    });
-  </script>
-</dom-module>
+    <template>
+        <style>
+             :host {
+                display: block;
+            }
+            kano-display-training {
+                margin: auto;
+                max-width: 700px;
+            }
+        </style>
+
+        <template is="dom-repeat" items="[[challengeSets]]" as="challengeSet">
+
+            <kwc-challenge-set
+                label="Challenge Set"
+                challenge-set="[[challengeSet]]"
+                progress="[[_computeCategoryProgress(challengeSet.category, completedChallenges)]]"
+                avatar-url="[[_userAvatarUrl]]"
+            >
+            </kwc-challenge-set>
+
+        </template>
+
+    </template>
+
+    <script>
+        Polymer({
+
+            is: 'kwc-challenge-set-listing',
+
+            properties: {
+                /*
+                 * An array of challenge set objects. A <kwc-challenge-set> element is rendered for each set.
+                 */
+                challengeSets: {
+                    type: Array,
+                    value: () => {
+                        return [];
+                    }
+                },
+                /*
+                 * An url for the current user's avatar
+                 */
+                userAvatarUrl: {
+                    type: String,
+                    value: null
+                },
+                _userAvatarUrl: {
+                    type: String,
+                    computed: '_computeAvatar(userAvatarUrl)'
+                },
+                /*
+                 * A dictionary of challenge categories containing an array of challenge ids that have been completed for each category.
+                 */
+                completedChallenges: {
+                    type: Object,
+                    value: () => {
+                        return {};
+                    }
+                },
+            },
+            _computeAvatar (userAvatarUrl) {
```

Comment:
Same as the comments on the other PR, it would be nice to store those assets on a separated file just for keeping this one easy to browse and sexy for our blue eyes u.u